crates/bevy_ui/src/render/mod.rs

```diff
@@ -177,6 +179,52 @@ pub struct ExtractedUiNodes {
     pub uinodes: Vec<ExtractedUiNode>,
 }
 
+pub fn extract_uinode_borders(
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    ui_stack: Extract<Res<UiStack>>,
+    uinode_query: Extract<
+        Query<(
+            &GlobalTransform,
+            &CalculatedBorder,
+            &BorderStyle,
+            &ComputedVisibility,
+            Option<&CalculatedClip>,
+        )>,
+    >,
+) {
+    for (stack_index, entity) in ui_stack.uinodes.iter().enumerate() {
+        if let Ok((transform, border, border_style, visibility, clip)) = uinode_query.get(*entity) {
+            // Skip invisible nodes
+            if !visibility.is_visible() || border_style.color.a() == 0.0 {
+                continue;
+            }
+
+            let image = DEFAULT_IMAGE_HANDLE.typed();
+
+            for &border_rect in border.edges.iter().flatten() {
+                let center = border_rect.center();
+                let transform = Mat4::from_translation(Vec3::new(center.x, center.y, 0.0))
+                    * transform.compute_matrix();
```

Comment:
is this expensive? could it be done one loop level higher?

Reply:
Each of the four rects making up the border is in a different position, so the translation has to be done in the inner loop I think. 

Perhaps there is a more efficient way to do the calculation. Maybe something like:
```rust
let mut transform = transform.affine();
transform.translation += Vec3A::new(center.x, center.y, 0.0);    
```
I don't know, I'll run some benchmarks on a few different methods.

Reply:
I was thinking that the `transform.compute_matrix()` should be the same for all sides

Reply:
`compute_matrix` just copies the transform column by column into a `Mat4`, so I don't think it's very expensive.